internal/helpers/deployment.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package helpers
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	"github.com/hashicorp/vault-secrets-operator/internal/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DeploymentStatusConfigMapName             = "deploymentstatus"
+	DeploymentStatusDeletionTimestampReceived = "deletion_timestamp_received"
+	DeploymentStatusManagerAcked              = "manager_acked"
+	StringTrue                                = "true"
+	StringFalse                               = "false"
+)
+
+type deploymentStatus map[string]string
+
+func CreateDeploymentStatusConfigMap(ctx context.Context, c client.Client) error {
+	status := deploymentStatus{
+		DeploymentStatusDeletionTimestampReceived: StringFalse,
+		DeploymentStatusManagerAcked:              StringFalse,
+	}
+
+	configMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DeploymentStatusConfigMapName,
+			Namespace: common.OperatorNamespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: common.OperatorDeploymentAPIVersion,
+					Kind:       common.OperatorDeploymentKind,
+					Name:       common.OperatorDeploymentName,
+					UID:        common.OperatorDeploymentUID,
+				},
+			},
+		},
+		Data: status,
+	}
+	if err := c.Create(ctx, &configMap); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func AwaitManagerAcked(ctx context.Context, logger logr.Logger, c client.Client) {
+	var (
+		configMap corev1.ConfigMap
+		key       = getDeploymentStatusConfigMapKey()
+	)
+
+	for {
+		select {
+		case <-ctx.Done():
+			logger.Error(ctx.Err(), "failed to await manager acked")
+			return
+		default:
+			if err := c.Get(ctx, key, &configMap); err != nil {
```

Comment:
It requires a slightly expanded client interface, but the k8s client package has a `WithWatch` interface that wraps the `client.Client` currently available which could be used to more efficiently wait for updates to the config map: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#WithWatch. Happy to pair up on helping you get going with that, as the documentation isn't super detailed.